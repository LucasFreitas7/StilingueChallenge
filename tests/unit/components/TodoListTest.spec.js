import { shallowMount, createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'
import TodoList from '../../../src/components/TodoList.vue'

const localVue = createLocalVue()

localVue.use(Vuex)

const handleClickOnFilterMock = jest.fn()
const clearDoneMock = jest.fn()
describe('TodoBar.vue', () => {
  let getters
  let store

  beforeEach(() => {
    getters = {
      doneAll: () => [1, 2, 3],
      undoneAll: () => [1, 2, 3]
    }
    store = new Vuex.Store({
        getters
    })
  })
  it('teste de snapshot', () => {
    const wrapper = shallowMount(TodoList, { store, localVue })
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('Quando clicado no All, deve ser disparado o evento de filtro', () => {
    const wrapper = shallowMount(TodoList, { store, localVue })
    wrapper.setMethods({handleClickOnFilter: handleClickOnFilterMock})
    const input = wrapper.find('a').trigger('click')
    expect(handleClickOnFilterMock).toHaveBeenCalled()
  })
  it('Quando clicado no Active, deve ser disparado o evento de filtro', () => {
    const wrapper = shallowMount(TodoList, { store, localVue })
    wrapper.setMethods({handleClickOnFilter: handleClickOnFilterMock})
    const input = wrapper.findAll('a').at(1).trigger('click')
    console.log(wrapper.html())
    expect(handleClickOnFilterMock).toHaveBeenCalled()
  })
  it('Quando clicado no Completed, deve ser disparado o evento de filtro', () => {
    const wrapper = shallowMount(TodoList, { store, localVue })
    wrapper.setMethods({handleClickOnFilter: handleClickOnFilterMock})
    const input = wrapper.findAll('a').at(2).trigger('click')
    expect(handleClickOnFilterMock).toHaveBeenCalled()
  })
  it('Quando clicado no Clear Completed, deve ser disparado a função ClearDone', () => {
    const wrapper = shallowMount(TodoList, { store, localVue })
    wrapper.setMethods({clearDone: clearDoneMock})
    const input = wrapper.findAll('a').at(3).trigger('click')
    expect(clearDoneMock).toHaveBeenCalled()
  })
})
