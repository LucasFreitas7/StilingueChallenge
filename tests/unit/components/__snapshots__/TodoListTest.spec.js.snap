// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoBar.vue teste de snapshot 1`] = `
<div class="list-container shadow">
  <div class="btn-container"><span class="left-items">3 items left</span>
    <div class="btn-filters"><a class="active">All</a> <a class="">Active</a> <a class="">Completed</a></div>
    <div class="btn-clear"><a>Clear completed</a></div>
  </div>
  <div class="btn-container-mobile">
    <div class="btn-clear-mobile"><a>Clear completed</a></div>
  </div>
</div>
`;
